zephyr/src/supp_api.c

```diff
@@ -165,100 +298,48 @@ int zephyr_supp_connect(const struct device *dev,
 		goto out;
 	}
 
-	wpa_supplicant_remove_all_networks(wpa_s);
-
-	ssid = wpa_supplicant_add_network(wpa_s);
-	ssid->ssid = os_zalloc(sizeof(u8) * MAX_SSID_LEN);
-
-	memcpy(ssid->ssid, params->ssid, params->ssid_length);
-	ssid->ssid_len = params->ssid_length;
-	ssid->disabled = 1;
-	ssid->key_mgmt = WPA_KEY_MGMT_NONE;
-
-	if (params->channel != WIFI_CHANNEL_ANY) {
-		/* We use global channel list here and also use the widest
-		 * op_class for 5GHz channels as there is no user input
-		 * for these.
-		 */
-		int freq  = ieee80211_chan_to_freq(NULL, 81, params->channel);
-
-		if (freq <= 0) {
-			freq  = ieee80211_chan_to_freq(NULL, 128, params->channel);
-		}
-
-		if (freq <= 0) {
-			wpa_printf(MSG_ERROR, "Invalid channel %d", params->channel);
-			ret = -EINVAL;
-			goto out;
-		}
-
-		ssid->scan_freq = os_zalloc(2 * sizeof(int));
-		if (!ssid->scan_freq) {
-			ret = -ENOMEM;
-			goto out;
-		}
-		ssid->scan_freq[0] = freq;
-		ssid->scan_freq[1] = 0;
-
-		ssid->freq_list = os_zalloc(2 * sizeof(int));
-		if (!ssid->freq_list) {
-			os_free(ssid->scan_freq);
-			ret = -ENOMEM;
-			goto out;
-		}
-		ssid->freq_list[0] = freq;
-		ssid->freq_list[1] = 0;
+	call_wpa_cli("remove_network all");
+	ret = wpa_cli_api_add_network(&resp);
+	if (ret) {
+		goto out;
 	}
 
-	wpa_s->conf->filter_ssids = 1;
-	wpa_s->conf->ap_scan = 1;
+	wpa_printf(MSG_DEBUG, "NET added: %d\n", resp.network_id);
 
+	call_wpa_cli("set_network %d ssid \"%s\"", resp.network_id, params->ssid);
+	call_wpa_cli("set_network %d key_mgmt NONE", resp.network_id);
 	if (params->psk) {
-		// TODO: Extend enum wifi_security_type
+		/* WPA3 */
 		if (params->security == 3) {
-			ssid->key_mgmt = WPA_KEY_MGMT_SAE;
-			str_clear_free(ssid->sae_password);
-			ssid->sae_password = dup_binstr(params->psk, params->psk_length);
-
-			if (ssid->sae_password == NULL) {
-				wpa_printf(MSG_ERROR, "%s:Failed to copy sae_password\n",
-					      __func__);
-				ret = -ENOMEM;
-				goto out;
+			if (params->sae_password) {
```

Comment:
Don't forget to set key_mgmt: `call_wpa_cli("set_network %d key_mgmt SAE", resp.network_id);`